0032-peering-clearing-settlement/0032-peering-clearing-settlement.md

```diff
@@ -0,0 +1,160 @@
+
+In the latter case, the sending peer may need to throttle back on the packets it sends to the upstream peer or they will need to make an alternatiuve settlement arrangement that can accomodate the volume.
+
+## Settlement Threshold
+
+When a connector receives a successful response from a peer it will evaluate if the amount of the packet, subtracted from the current balance, is less than the settlement threshold. If so, the connector should perform a settlement, following which the balance on the Interledger account is adjusted up by the settlement amount.
+
+In a correctly configured peering the additive inverse (negation) of the settlement threshold of one peer will be less than the maximum balance of the other peer.
+
+> **Example : Alice and Bob**
+
+> Assuming they start with a balance of 0 and have configured the following. 
+
+> - Alice has configured a settlement threshold of -10. This means Alice will settle with Bob as soon as she owes him 10 satoshis or more.
+> - Alice has configured a maximum balance of 20. This means Alice will reject any packets from Bob that would move the balance higher than 20.
+> - Bob has configured a settlement threshold of -15. This means Bob will settle with Alice as soon as he owes her 15 satoshis or more.
+> - Bob has configured a maximum balance of 5. This means Bob will reject any packets from Alice that would move the balance higher than 5 satoshis.
+
+> If the following sequence of events occurs:
+
+> 1. Bob passes an ILP Prepare to Alice with an amount of 12.
+> 1. Alice replies with a valid ILP Fulfill. Alice's balance is now 12. Bob's balance is -12.
+> 1. Bob passes an ILP Prepare to Alice with an amount of 9.
+> 1. Alice rejects this with a T04 error code because this would take her balance over 20.
+> 1. Bob passes an ILP Prepare to Alice with an amount of 7.
+> 1. Alice replies with a valid ILP Fulfill. Alice's balance is now 19. Bob's balance is -19.
+> 1. Bob's balance is now past his settlement threshold of -15 so he makes a 19 satoshi Lightning payment to Alice and notifies her that he has settled the account. Bob's is now 0.
+> 1. Alice receives the payment and adjusts her balance to 0.
+> 1. Alice passes an ILP Prepare to Bob with an amount of 2.
+> 1. Bob replies with a valid ILP Fulfill. Bob's balance is now 2. Alice's balance is -2.
+> 1. Alice passes an ILP Prepare to Bob with an amount of 5.
+> 1. Bob rejects this with a T04 error code because this would take his balance over 5.
+
+> Note how, because Bob has used very conservative limits on his maximum balance, and Alice has used a larger number for her settlement threshold, a situation will regularly arise where Bob reject packets from Alice because she is going to exceed Bob's maximum balance but she never hits her settlement threshold. While Alice can overcome this by settling whenever Bob rejects a packet with a T04 error a better solution would be for Alice to configure her threshold to be within Bob's maximum balance.
```

Comment:
It may also be worth mentioning using F08 errors to indicate the maximum allowable packet size. That can help tell senders how much they should send in each packet, which can help avoid this situation. Or we could leave this out and just suggest that these two should raise their limits with one another

Reply:
I think we need another whole doc explaining throughput/bandwidth